src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -12,37 +13,28 @@ namespace System.Data.SqlClient.SNI
     /// <summary>
     /// SNI Packet
     /// </summary>
-    internal partial class SNIPacket : IDisposable, IEquatable<SNIPacket>
+    internal sealed partial class SNIPacket
     {
-        private byte[] _data;
-        private int _length;
-        private int _capacity;
-        private int _offset;
-        private string _description;
-        private SNIAsyncCallback _completionCallback;
-        private bool _isBufferFromArrayPool;
-
-        public SNIPacket() { }
-
-        public SNIPacket(int capacity)
+        [Flags]
+        private enum SNIPacketFlags : uint
         {
-            Allocate(capacity);
+            None = 0,
+            ArrayFromPool = 1,
+            MuxHeaderReserved = 2,
+            MuxHeaderWritten = 4,
         }
 
-        /// <summary>
-        /// Packet description (used for debugging)
-        /// </summary>
-        public string Description
-        {
-            get
-            {
-                return _description;
-            }
+        private int _length; // the length of the data in the data segment, advanced by Append-ing data, does not include smux header length
+        private int _capacity; // the total capacity requested, if the array is rented this may be less than the _data.Length, does not include smux header length
+        private int _offset; // the start point of the data in the data segment, advanced by Take-ing data
+        private int _header; // the amount of space at the start of the array reserved for the smux header, this is zeroed in SetHeader
```

Comment:
THe name `_header` sounds more like data than a number. Can you rename this to `_headerSize`

Reply:
Renamed throughout. headerLength, dataLength, dataOffset, dataCapacity. It reads better to me, see if you agree or have other naming suggestions.